cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -226,7 +234,64 @@ func (ng *nodegroup) Nodes() ([]cloudprovider.Instance, error) {
 // node by default, using manifest (most likely only kube-proxy).
 // Implementation optional.
 func (ng *nodegroup) TemplateNodeInfo() (*schedulerframework.NodeInfo, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	if !ng.scalableResource.CanScaleFromZero() {
+		return nil, cloudprovider.ErrNotImplemented
+	}
+
+	capacity, err := ng.scalableResource.InstanceCapacity()
+	if err != nil {
+		return nil, err
+	}
+
+	_, cpuPresent := capacity[corev1.ResourceCPU]
+	_, memPresent := capacity[corev1.ResourceMemory]
+	if !cpuPresent || !memPresent {
+		return nil, cloudprovider.ErrNotImplemented
```

Comment:
is `cloudprovider.ErrNotImplemented` the right error type to return for this error condition?

Reply:
After reading through [this](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20210310-opt-in-autoscaling-from-zero.md#implementation-detailsnotesconstraints) it seems like the answer is yes.

Reply:
Actually now I see that the `CanScaleFromZero()` method already validates the presence of cpu and memory capacity:

https://github.com/kubernetes/autoscaler/pull/4840/files#diff-2052c87a1cd8247c67b54b3e35acc9dff157210e845256b2282f87ac46bf805fR205

So I think we can remove L246-250 entirely

Reply:
removed the extra lines and added some code docs